src/agent/communicator/src/communicator.cpp

```diff
@@ -101,46 +101,59 @@ namespace communicator
             }();
 
             m_tokenExpTimer->expires_after(duration);
-            co_await m_tokenExpTimer->async_wait(boost::asio::use_awaitable);
+
+            boost::system::error_code ec;
+            co_await m_tokenExpTimer->async_wait(boost::asio::redirect_error(boost::asio::use_awaitable, ec));
+
+            if (ec == boost::asio::error::operation_aborted)
```

Comment:
Maybe we can put this if inside the next one?

Reply:
Yes, I've also removed the `continue`. It wasn't necessary. This is just about printing the correct error message.